types/servicelimits.pp

```diff
@@ -7,7 +7,7 @@
     Optional['LimitSTACK']          => Pattern['^(infinity|((\d+(K|M|G|T|P|E)(:\d+(K|M|G|T|P|E))?)))$'],
     Optional['LimitCORE']           => Pattern['^(infinity|((\d+(K|M|G|T|P|E)(:\d+(K|M|G|T|P|E))?)))$'],
     Optional['LimitRSS']            => Pattern['^(infinity|((\d+(K|M|G|T|P|E)(:\d+(K|M|G|T|P|E))?)))$'],
-    Optional['LimitNOFILE']         => Integer[0],
+    Optional['LimitNOFILE']         => Pattern['^(infinity|(\d+))$'],
```

Comment:
Is ```Variant[Integer[0], Enum['infinity']]``` cleaner? This changes the datatype from a real integer to a string representation of integers.

Reply:
Agree, my commit would break current usage, my bad. Let me change it to your proposal, is exactly what im looking for, just add for infinity. Thanks!